stack/globals_test.go

```diff
@@ -1103,6 +1107,254 @@ func TestLoadGlobals(t *testing.T) {
+					),
+				},
+			},
+			want: map[string]*hclwrite.Block{
+				"/stack": globals(
+					attr("team", `{ def = { name = "awesome" } }`),
+				),
+			},
+		},
+		{
+			name: "globals from imported file and merging",
+			layout: []string{
+				"d:other",
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path:     "/stack",
+					filename: "globals.tm",
+					add: hcldoc(
+						importy(
+							attr("source", `"/other/globals.tm"`),
+						),
+						globals(
+							attr("team2", `"test"`),
+						),
+					),
+				},
+				{
+					path:     "/other",
+					filename: "globals.tm",
+					add: globals(
+						attr("team", `{ def = { name = "awesome" } }`),
+					),
+				},
+			},
+			want: map[string]*hclwrite.Block{
+				"/stack": globals(
+					attr("team", `{ def = { name = "awesome" } }`),
+					attr("team2", `"test"`),
+				),
+			},
+		},
+		{
+			name: "redefined globals from imported file - fails",
+			layout: []string{
+				"d:other",
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path:     "/stack",
+					filename: "globals.tm",
+					add: hcldoc(
+						importy(
+							attr("source", `"/other/globals.tm"`),
+						),
+						globals(
+							attr("team", `{ def = { name = "redefined" } }`),
+						),
+					),
+				},
+				{
+					path:     "/other",
+					filename: "globals.tm",
+					add: globals(
+						attr("team", `{ def = { name = "defined" } }`),
+					),
+				},
+			},
+			wantErr: errors.E(hcl.ErrTerramateSchema),
+		},
+		{
+			name: "globals can reference imported values",
+			layout: []string{
+				"d:other",
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path:     "/stack",
+					filename: "cfg.tm",
+					add: hcldoc(
+						importy(
+							attr("source", `"/other/imported.tm"`),
+						),
+						globals(
+							expr("B", `"redefined from ${global.A}"`),
+						),
+					),
+				},
+				{
+					path:     "other",
+					filename: "imported.tm",
+					add: globals(
+						attr("A", `"imported"`),
+					),
+				},
+			},
+			want: map[string]*hclwrite.Block{
+				"/stack": globals(
+					attr("B", `"redefined from imported"`),
+					attr("A", `"imported"`),
+				),
+			},
+		},
+		{
+			name: "imported files are handled before importing file",
+			layout: []string{
+				"d:other",
+				"s:stack",
+			},
+			configs: []hclconfig{
+				{
+					path:     "/stack",
+					filename: "cfg.tm",
+					add: hcldoc(
+						globals(
+							expr("B", `"redefined from ${global.A}"`),
+						),
+						importy(
+							attr("source", `"/other/imported.tm"`),
+						),
+					),
+				},
+				{
+					path:     "other",
+					filename: "imported.tm",
+					add: globals(
+						attr("A", `"other/imported.tm"`),
+					),
+				},
+			},
+			want: map[string]*hclwrite.Block{
+				"/stack": globals(
+					attr("B", `"redefined from other/imported.tm"`),
+					attr("A", `"other/imported.tm"`),
+				),
+			},
+		},
+
+		{
+			name: "imported file has redefinition of own imports",
```

Comment:
The way redefinition is being used accross the tests is confusing to me, nothing is being redefined here AFAIK, you have different globals being defined on different levels/scopes using values of other scopes. Maybe I'm just getting redefinition wrong ? :thinking: but for me it would be something like:

parent:

```
global {
  a = 1
}
```

Child:

```
global {
  a = 2
}
```

In the child config, global.a was redefined from 1 to 2. Creating a new global given the value of a different one is just a definition with a reference to another global no ? :thinking:

Reply:
What is happening here is more like "imports definitions can reference other import definitions" IMHO.

Reply:
fixed this. Sorry...